### PR TITLE
Issue #3: Make the nav bar transition smooth and intuitive 

### PR DESCRIPTION
**Pull request by genial-mani**
This is the final result after fixing the issue.

localhost_3000_-GoogleChrome2024-10-0300-23-43-ezgif com-crop

if you like it please merge the PR.

### CODE REVIEW

**Comment by genial-mani** [comment]
could you please add hacktoberfest topic to the repo or to the pr please.

**Comment by g4ze** [comment]
great work! can you pls follow the PR template for an easy merge?

**Comment by genial-mani** [comment]
Yes, I already followed the same:

**Important Checklist:**

1.  You have prettyfied your code (npx prettier --write .)
2.  Linted the code using ESLint (npx eslint . --ext .js,.jsx,.ts,.tsx)
3.  This code is best you could write in terms of quality (we don't mind even if its bad :P)
4.  You have attached the screenshot of visible chnages in GUI, if any
5.  You have no merge conflicts, you have pulled and rebased your PR branch
6.  All CI tests have passed


also if you have any queries bring them to the table.
if you don't have any queries merge the PR after adding hacktoberfest topic to the repo/PR.

**Review by g4ze — CHANGES_REQUESTED** [review]

File: src/app/page.js [hunk 1]
```diff
@@ -1,75 +1,83 @@
...
+        });
+      }, 600);
     }
   };
 
   return (
     <>
-      <nav class="fixed w-full z-20 top-0 start-0  ">
-        <div class="max-w-screen-xl flex flex-wrap items-center justify-between mx-auto p-4">
+      <nav className={`fixed w-full z-20 top-0 start-0  transition-all duration-300 ease-in-out ${visible ? "transform translate-y-0" : "transform -translate-y-full"}`}>
```
[the remark this answers is not in the document]
Comment (genial-mani) [reply src/app/page.js]: I resolved this. If you have any other questions, please specify and add hacktoberfest topic to the repo or PR before merging.

File: src/components/Navbar.jsx [hunk 2]
```diff
@@ -1,38 +1,67 @@
-export default Navebar = () => {
+import { useEffect, useState } from "react";
+import Image from "next/image";
+
+
+export default function  Navbar () {
+  const [visible, setVisible] = useState(false);
+  const [scrollPosition, setScrollPosition] = useState(0);
+
+  useEffect(() => {
```
Comment (g4ze) [inline src/components/Navbar.jsx:9]: i think this function is not supposed to be here, can you pls specify any particular reasons for not keeping in `page.js` and using it here?

**Review by g4ze — CHANGES_REQUESTED** [review]

File: src/components/Navbar.jsx (hunk 2 above)
Comment (genial-mani) [reply src/components/Navbar.jsx:9]: which function you want to know about please mention.

Comment (g4ze) [reply src/components/Navbar.jsx:9]: Useeffect

Comment (genial-mani) [reply src/components/Navbar.jsx:9]: UseEffect is used for the navbar and only in the Navbar.jsx component no need every where.

**Pull request merged**